app/models/session.rb

```diff
@@ -182,8 +182,8 @@ def vaccine_methods
     programmes.flat_map(&:vaccine_methods).uniq.sort
   end
 
-  def programmes_for(patient: nil, year_group: nil)
-    year_group ||= patient.year_group
+  def programmes_for(year_group: nil, patient: nil, academic_year: nil)
+    year_group ||= patient.year_group(academic_year:)
```

Comment:
I don't see any tests for this method. Do you reckon the existing testing covers it? Or could there be some edge-cases, especially with the addition of `academic_year`, that haven't been covered by other tests?

Reply:
I think it's covered by existing tests for the current academic year, tests for upcoming academic years will be added in #2712.

Reply:
I linked to the wrong PR originally, have fixed the reference to one that isn't already merged in 😆